util/dvsim/LintCfg.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+r"""
+Class describing lint configuration object
+"""
+
+import logging as log
+import sys
+from pathlib import Path
+
+from tabulate import tabulate
+
+from .Deploy import *
+from .Modes import *
+from .OneShotCfg import OneShotCfg
+from .utils import *
+
+
+# helper function for printing messages
+def _print_msg_list(msg_list_name, msg_list):
+    md_results = ""
+    if msg_list:
+        md_results += "### %s\n" % msg_list_name
+        md_results += "```\n"
+        for msg in msg_list:
+            msg_parts = msg.split()
+            md_results += msg + "\n\n"
+        md_results += "```\n"
+    return md_results
+
+
+class LintCfg(OneShotCfg):
+    """Derivative class for linting purposes.
+    """
+    def __init__(self, flow_cfg_file, proj_root, args):
+        super().__init__(flow_cfg_file, proj_root, args)
+
+    @staticmethod
+    def create_instance(flow_cfg_file, proj_root, args):
+        '''Create a new instance of this class as with given parameters.
+        '''
+        return LintCfg(flow_cfg_file, proj_root, args)
+
+    def gen_results_summary(self):
+        '''
+        Gathers the aggregated results from all sub configs
+        '''
+
+        # Generate results table for runs. Note that we build a simple md and
+        # a marked up md version here in parallel.
+        log.info("Create summary of lint results")
+
+        results_str = "## Lint Summary Results\n\n"
+        results_str += "### " + self.timestamp_long + "\n\n"
+
+        header = [
+            "Name", "Tool Warnings", "Tool Errors", "Lint Warnings",
+            "Lint Errors"
+        ]
+        colalign = ("center", ) * len(header)
+        table = [header]
+
+        num_result_rows = 0
+        for cfg in self.cfgs:
+
+            results_page = cfg.results_server_dir + '/results.html'
+            results_page_url = results_page.replace(
+                cfg.results_server_prefix, cfg.results_server_url_prefix)
+            name_with_link = "[" + cfg.name.upper(
+            ) + "](" + results_page_url + ")"
+            table.append([
+                name_with_link,
+                str(len(cfg.result_summary["warnings"])) + " W",
+                str(len(cfg.result_summary["errors"])) + " E",
+                str(len(cfg.result_summary["lint_warnings"])) + " W",
+                str(len(cfg.result_summary["lint_errors"])) + " E"
+            ])
+
+            num_result_rows += 1
+
+        if num_result_rows:
```

Comment:
can we do this instead: `if len(table) > 1:`? Then no need to maintain this var.

Reply:
sure, will do